internal/plugin/connectors/tcp/mssql/connection_details.go

```diff
@@ -63,3 +74,17 @@ func NewConnectionDetails(credentials map[string][]byte) *ConnectionDetails {
 func (cd *ConnectionDetails) Address() string {
 	return cd.Host + ":" + strconv.FormatUint(uint64(cd.Port), 10)
 }
+
+func (cd *ConnectionDetails) parseSSLOptions(credentials map[string][]byte) {
+	cd.SSLOptions = make(map[string]string)
+
+	for _, sslOption := range sslOptions {
+		if len(credentials[sslOption]) > 0 {
```

Comment:
Ditto: this might panic if that field isn't present in the config (which ends up in the credentials)